bundles/io/org.eclipse.smarthome.io.net.test/src/test/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactoryTest.java

```diff
@@ -0,0 +1,196 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Stream;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.smarthome.io.net.http.HttpClientFacade;
+import org.eclipse.smarthome.io.net.http.TrustManagerProvider;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.Mock;
+
+public class SecureHttpClientFactoryTest {
+
+    private SecureHttpClientFactory secureHttpClientFactory;
+
+    @Mock
+    private TrustManagerProvider trustmanagerProvider;
+
+    @Before
+    public void setup() {
+        initMocks(this);
+        secureHttpClientFactory = new SecureHttpClientFactory();
+        secureHttpClientFactory.setTrustmanagerProvider(trustmanagerProvider);
+    }
+
+    @Test
+    public void testGetClient() throws Exception {
+        secureHttpClientFactory.activate(createConfigMap(10, 200, 60, 5, 10, 60));
+
+        HttpClientFacade client = secureHttpClientFactory.getHttpClient();
+
+        assertThat(client, is(notNullValue()));
+
+        secureHttpClientFactory.deactivate();
+    }
+
+    @Test
+    public void testGetClientWithEndpoint() throws Exception {
+        secureHttpClientFactory.activate(createConfigMap(10, 200, 60, 5, 10, 60));
+
+        when(trustmanagerProvider.getTrustManagers("https://www.heise.de")).thenReturn(Stream.empty());
+
+        HttpClient client = secureHttpClientFactory.createHttpClient("consumer", "https://www.heise.de");
+
+        assertThat(client, is(notNullValue()));
+
+        verify(trustmanagerProvider).getTrustManagers("https://www.heise.de");
+
+        client.stop();
+
+        secureHttpClientFactory.deactivate();
+    }
+
+    @Ignore("only for manual test")
+    @Test
+    public void testMultiThreadedShared() throws Exception {
+        secureHttpClientFactory.activate(createConfigMap(10, 200, 60, 5, 10, 60));
+
+        ThreadPoolExecutor workers = new ThreadPoolExecutor(20, 80, 60, TimeUnit.SECONDS,
+                new ArrayBlockingQueue<Runnable>(50 * 50), new NamedThreadFactory("workers"));
+
+        List<HttpClientFacade> clients = new ArrayList<>();
+
+        final int CLIENTS = 2;
+        final int REQUESTS = 2;
+
+        for (int i = 0; i < CLIENTS; i++) {
+            HttpClientFacade httpClient = secureHttpClientFactory.getHttpClient();
+            clients.add(httpClient);
+        }
+
+        final List<String> failures = new ArrayList<>();
+
+        for (int i = 0; i < REQUESTS; i++) {
+            for (final HttpClientFacade client : clients) {
+                workers.execute(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            ContentResponse response = client.GET("https://www.heise.de");
```

Comment:
Not sure if they like receiving the requests from our tests. Maybe www.eclipse.org would be the better target?

Reply:
I will change this. In any way these tests are not intended to be executed automatically.